assets/js/scripts.js

```diff
@@ -33,8 +33,34 @@ $(function(){
             var siteUrl = siteHeader.href.toLowerCase();
             var lowerTerm = term.toLowerCase();
 
-            // returns true if lowerTerm isn't found in site title or URL
-            return Math.max(siteTitle.indexOf(lowerTerm), siteUrl.indexOf(lowerTerm)) === -1;
+            // split words at spaces/dashes (' ' and '-')
+            const splitTerm = lowerTerm.trim().split(/ |-/g);
+            const splitTitle = siteTitle.trim().split(/ |-/g);
+            const siteDomain = siteUrl.match(/(?<=http(?:s?):\/\/)[^\/]+/)[0];
+           
+            // don't hide if substantial part (length > 3) of term is included in title
+            for (let i = 0; i < splitTerm.length; i++) {
+               if (siteTitle.indexOf(splitTerm[i]) != -1 && splitTerm[i].length > 3) {
+                 return false;
+               }
+            }
+            // don't hide if substantial part (length > 3) of title is included in term
+            for (let i = 0; i < splitTitle.length; i++) {
+                if (lowerTerm.indexOf(splitTitle[i]) != -1 && splitTitle[i].length > 3) {
+                  return false;
+                }
+            }
+            // don't hide if term is included in url
+            if (siteDomain.indexOf(lowerTerm) != -1) {
+                return false;
+            }
+            // if other options fail, default to old search
+            if (Math.max(siteTitle.indexOf(lowerTerm), siteUrl.indexOf(lowerTerm)) != -1) {
+                return false;
+            }
+           
+            // else, hide
+            return true;
```

Comment:
Regardless of my earlier comment, given that the old search is the "default", it does not make sense for there to be a `return true;` line following that case. That would be like writing a switch statement where a `case:` is expected to be the default, but a separate `default:` block is still provided.